var/spack/repos/builtin/packages/glusterfs/package.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Glusterfs(AutotoolsPackage):
+    """Gluster is a software defined distributed storage that can scale to
+    several petabytes. It provides interfaces for object, block and file
+    storage."""
+
+    homepage = "https://www.gluster.org/"
+    url      = "https://download.gluster.org/pub/gluster/glusterfs/7/7.3/glusterfs-7.3.tar.gz"
+    list_depth = 2
```

Comment:
You still need a `list_url`: https://spack.readthedocs.io/en/latest/packaging_guide.html#attribute-list-url

Reply:
I have overwrited url_for_version. 
maybe I should just remove `list_depth` and no need for `list_url` definition?
```
    def url_for_version(self, version):
        url = 'https://download.gluster.org/pub/gluster/glusterfs/{0}/{1}/glusterfs-{1}.tar.gz'
        return url.format(version.up_to(1), version)
```
I can stage all versions between 6.7 to 7.3 by using `url_for_version()`

Reply:
`url_for_version` is used to download a specific version (`spack fetch` and `spack install`)

`list_url` and `list_depth` are used to locate new versions (`spack versions` and `spack checksum`)

Both are needed in this case.

Reply:
Got it, thanks for your guidance.
Fix it in https://github.com/spack/spack/pull/15565/commits/5dc402712cbc5f30a64536c62b193154ddfd08b5